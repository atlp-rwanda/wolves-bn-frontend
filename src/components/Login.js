import React, { useState, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link, Redirect } from 'react-router-dom';
import { login } from '../redux/actions/login';
import InputField from './common/InputField';
import Button from './common/Button';
import logo from '../assets/undraw_travelers.svg';
import './login.scss';

const Login = () => {
  const newState = useSelector(state => state);
  console.log(newState);
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [submitted, setSubmitted] = useState(false);
  const dispatch = useDispatch();

  useEffect(() => {
    const { token, isLoggedIn } = newState.login;
    if (isLoggedIn && token) {
      localStorage.setItem('token', token);
      window.location.replace('/dashboard');
    }
  });

  return (
        <div className='container'>
          <div className='row content'>
            <div className='col-md-6 mb-3'>
              <img src={logo} alt='' className='img-fluid' />
            </div>
            <div className='col-md-6'>
              <h3 className='signin-text mb-3'>Sign In</h3>
              <form className='loginForm' action='#' onSubmit={(e) => {
                e.preventDefault();
                setSubmitted(true);

                if (email && password) {
                  dispatch(login(email, password));
                }
              }}>
                {submitted && newState.login.message && <div className= 'invalid-feedback'>{newState.login.message}</div> }
                <div className='form-group'>
                  <InputField
                  type='email'
                  name='email'
                  label='Email'
                  className='form-control'
                  onChange = {(e) => { setEmail(e.target.value); }}
                  />
                  {submitted && !email && <div className="invalid-feedback">Email is required</div>}
                </div>
                <div className='form-group'>
                  <InputField
                  label='Password'
                  type='password'
                  name='password'
                  className='form-control'
                  onChange = {(e) => { setPassword(e.target.value); }}
                  />
                  {submitted && !password
                    && <div className="invalid-feedback">Password is required</div>}
                </div>
                <div className='form-group'>
                  <ul className='expLinks'>
                    <li><Link to='/signup' ><h4><span>Create Account</span> </h4></Link></li>
                    <li> <Link to='/forgotpassword'><span>Forgot Password?</span></Link></li>
                  </ul>
                </div>
                <div className="form-group">
                <Button content='Login' />
            </div>
            </form>
          </div>
        </div>
      </div>
  );
};

export default Login;