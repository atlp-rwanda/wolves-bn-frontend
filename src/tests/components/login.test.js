import React from 'react';
import configureMockStore from 'redux-mock-store';
import promise from 'redux-promise-middleware';
import { BrowserRouter as Router } from 'react-router-dom';
import Enzyme, { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Login from '../../components/Login';

Enzyme.configure({ adapter: new Adapter() });

const middlewares = [thunk, promise];
const mockStore = configureMockStore(middlewares);

describe('Login Hooks', () => {
  let wrapper;
  const store = mockStore({});
  const handleSubmit = jest.fn;
  beforeEach(() => {
    wrapper = mount(
    <Router>
    <Provider store={store}>
    <Login handleSubmit={handleSubmit} />
    </Provider>
    </Router>);
  });
  store.dispatch = jest.fn();
});

describe('Login component', () => {
  const storeBfrLogin = mockStore({
    login: {
      isLoggedIn: false,
      token: null
    },
  });

  storeBfrLogin.dispatch = jest.fn();

  const component = renderer.create(
    <Router>
    <Provider store={storeBfrLogin}>
      <Login />
    </Provider>
    </Router>
  );

  it('should render with initial state', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});