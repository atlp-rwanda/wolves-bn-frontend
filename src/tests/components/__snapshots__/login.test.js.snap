// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component should render with initial state 1`] = `
<div
  className="loginPage"
>
  <div
    className="row content"
  >
    <div
      className="col-md-6 mb-3"
    >
      <img
        alt=""
        className="img-fluid"
        src="test-file-stub"
      />
    </div>
    <div
      className="col-md-6"
    >
      <h3
        className="signin-text mb-3"
      >
        Sign In
      </h3>
      <form
        action="#"
        className="loginForm"
        onSubmit={[Function]}
      >
        <div
          className="form-group"
        >
          <div
            className="inputDiv"
          >
            <label>
              Email
            </label>
            <input
              name="email"
              onChange={[Function]}
              type="email"
            />
          </div>
        </div>
        <div
          className="form-group"
        >
          <div
            className="inputDiv"
          >
            <label>
              Password
            </label>
            <input
              name="password"
              onChange={[Function]}
              type="password"
            />
          </div>
        </div>
        <div
          className="form-group"
        >
          <ul
            className="expLinks"
          >
            <li>
              <a
                href="/signup"
                onClick={[Function]}
              >
                <h4>
                  <span>
                    Create Account
                  </span>
                   
                </h4>
              </a>
            </li>
            <li>
               
              <a
                href="/forgotpassword"
                onClick={[Function]}
              >
                <span>
                  Forgot Password?
                </span>
              </a>
            </li>
          </ul>
        </div>
        <div
          className="form-group"
        >
          <button
            className="btn"
          >
            Login
          </button>
        </div>
      </form>
      <div>
        <div
          className="social"
        >
          <h3
            className="social-header"
          >
            Or sign in with: 
          </h3>
          <button
            className="social-btn google"
            onClick={[Function]}
            type="submit"
          >
            <img
              src="test-file-stub"
            />
          </button>
          <button
            className="social-btn facebook"
            onClick={[Function]}
            type="submit"
          >
            <img
              src="test-file-stub"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
